R/add_ppa_flag.R

```diff
@@ -0,0 +1,149 @@
+            "D501", "D508", "D509"
+          ) ~ TRUE,
+
+        # Reliant on any of the six diagnosis codes, first three characters
+        rowSums(dplyr::across(
+          c(.data$diag1, .data$diag2, .data$diag3, .data$diag4, .data$diag5, .data$diag6),
+          ~ stringr::str_sub(.x, 1, 3) %in%
+            c(
+              # Gangrene
+              "R02",
+              # Influenza
+              "J10", "J11", "J13",
+              # Vaccine-preventable
+              "A35", "A36", "A80", "B05", "B06", "B26"
+            )
+        )) > 0 ~ TRUE,
+
+        # Reliant on any of the six diagnosis codes, first four characters
+        rowSums(dplyr::across(
+          c(.data$diag1, .data$diag2, .data$diag3, .data$diag4, .data$diag5, .data$diag6),
+          ~ stringr::str_sub(.x, 1, 4) %in%
+            c(
+              # Vaccine-preventable
+              "A370", "A379", "B161", "B169",
+              # Diabetes
+              "E100", "E101", "E102", "E103", "E104",
+              "E105", "E106", "E107", "E108", "E110",
+              "E111", "E112", "E113", "E114", "E115",
+              "E116", "E117", "E118", "E120", "E121",
+              "E122", "E123", "E124", "E125", "E126",
+              "E127", "E128", "E130", "E131", "E132",
+              "E133", "E134", "E135", "E136", "E137",
+              "E138", "E140", "E141", "E142", "E143",
+              "E144", "E145", "E146", "E147", "E148",
+              # Influenza
+              "J181"
+            )
+        )) > 0 ~ TRUE,
+
+        # Reliant on op1a and diag1
+        # Angina
+        stringr::str_sub(.data$diag1, 1, 3) == "I20" &
+          !(stringr::str_sub(.data$op1a, 1, 3) %in% c("K40", "K45", "K49", "K60", "K65", "K66")) ~ TRUE,
+        # Cellulitis
+        stringr::str_sub(.data$diag1, 1, 3) %in% c("L03", "L04") &
+          !(stringr::str_sub(.data$op1a, 1, 3) %in% c("S06", "S57", "S68", "S70", "W90", "X11")) ~ TRUE,
+        stringr::str_sub(.data$diag1, 1, 4) %in% c("L080", "L088", "L089", "L980") &
+          !(stringr::str_sub(.data$op1a, 1, 3) %in% c("S06", "S57", "S68", "S70", "W90", "X11")) ~ TRUE,
+
+        # Reliant on diag1 and excluding_operation
+        stringr::str_sub(.data$diag1, 1, 3) %in% c(
+          # Angina
+          "I10",
+          # Congestive HF
+          "I50",
+          "J81"
+        ) &
+          excluding_operation == FALSE ~ TRUE,
+        stringr::str_sub(.data$diag1, 1, 4) %in% c(
+          # Hypertension
+          "I119", "I110"
+        ) &
+          excluding_operation == FALSE ~ TRUE,
+
+        # Reliant on diag1 and diag2
+        # Bronchitis
+        stringr::str_sub(.data$diag1, 1, 3) == "J40" &
+          stringr::str_sub(.data$diag2, 1, 3) %in% c("J41", "J42", "J43", "J44") ~ TRUE,
+
+        # Reliant on diag2 only
+        # Bronchitis
+        stringr::str_sub(.data$diag2, 1, 3) == "J47" ~ TRUE,
+
+        # All other values
+        TRUE ~ FALSE
+      )
+    )
+  return(matching_data)
```

Comment:
I think instead of returning this 'lookup style' data frame, it would be better to just do the join and return the full data.

At the moment the usage would be: `slf_data %>% left_join(add_ppa_flag(slf_data), by = c("anon_chi", "cij_marker"))` which is a bit ugly but if we did the join in the function.

e.g. 
```r
ppa_cij_data <- dplyr::left_join(data, matching_data, by = c("anon_chi", "cij_marker")) %>% 
   dplyr::rename(cij_ppa = .data$ppa)

return(ppa_cij_data)
```

Reply:
Added the join into the function and tested, works as expected